python_modules/libraries/dagster-aws/dagster_aws/ecs/launcher.py

```diff
@@ -10,6 +10,18 @@
 from dagster.grpc.types import ExecuteRunArgs
 from dagster.serdes import ConfigurableClass, serialize_dagster_namedtuple
 from dagster.utils.backcompat import experimental
+from dagster.utils.backoff import backoff
+
+
+# The ECS API is eventually consistent:
+# https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_RunTask.html
+# describe_tasks might initially return nothing even if a task exists.
+class EcsEventualConsistencyTimeout(Exception):
+    pass
+
+
+# 9 retries polls for up to 51.1 seconds with exponential backoff.
+BACKOFF_RETRIES = 9
```

Comment:
Should we bite the bullet and poll for the full 5 minutes that AWS recommends? If we do that, we'll probably want to leave some kind of breadcrumb in the event log to let users know why it's taking so long to launch.

Reply:
Up to you. I think we should definitely reporting some engine event if the eventual consistency timeout occurs though.

Reply:
What happens if we only raise? Will that not show up as an engine event? I sort of assumed it'd move the pipeline run from STARTED to FAILED but I haven't actually tried it.